containers/Dockerfile.buildenv

```diff
@@ -210,6 +210,12 @@ RUN git clone --single-branch --branch ${CHARM_GIT_TAG} --depth 1 \
     && git apply /work/charm/v6.8.patch \
     && rm /work/charm/v6.8.patch \
     && rm -r /work/charm/doc /work/charm/examples
+RUN git clone https://github.com/UIUC-PPL/charm charm_6_10_2 \
```

Comment:
ping on this suggestion https://github.com/sxs-collaboration/spectre/pull/2324#discussion_r439831813:

> Since we don't need to apply a patch anymore, perhaps better download the charm 6.10.1 release archive instead of git-cloning the repo? If you need to clone the repo, better download only the single branch as done above to reduce the size of the Docker image?

My suggestion: We merge this PR as-is, and when deleting Charm 6.8 from the container we can adjust this to reduce the size of the container, if needed. Perhaps open an issue to remember.